packages/synapse-react-client/src/components/GenericCard/TableRowGenericCard.tsx

```diff
@@ -0,0 +1,567 @@
+      >
+        (Show Version History on Synapse)
+      </a>
+    </span>
+  )
+}
+
+type ValueOrMultiValue = {
+  str: string
+  strList?: string[]
+  columnModelType?: ColumnType
+}
+
+export function getCardLinkHref(
+  cardLink: CardLink | undefined,
+  data: string[] | undefined,
+  schema?: Record<string, number>,
+  rowId?: number,
+): string | undefined {
+  if (cardLink) {
+    if (!data || !schema) {
+      throw Error('Must specify CardLink and data for linking to work')
+    }
+    const { matchColumnName, overrideValueWithRowID } = cardLink
+
+    // PORTALS-2088:  Return the link, unless...
+    // an overrideLinkURLColumnName has been set and its value is defined.
+    // In this case, just use the overrideLinkURLColumnName value
+    if ('overrideLinkURLColumnName' in cardLink) {
+      const { overrideLinkURLColumnName, overrideLinkURLColumnTransform } =
+        cardLink
+      if (schema[overrideLinkURLColumnName]) {
+        const indexOfOverrideLinkURLColumnName =
+          schema[overrideLinkURLColumnName]
+        const overrideLinkValue = data[indexOfOverrideLinkURLColumnName]
+        if (overrideLinkValue && overrideLinkURLColumnTransform) {
+          return overrideLinkURLColumnTransform(overrideLinkValue)
+        } else if (overrideLinkValue) {
+          return overrideLinkValue
+        }
+      }
+    }
+
+    const indexInData = schema[matchColumnName]
+    if (indexInData === undefined) {
+      console.error(
+        `Could not find match for data: ${data} with columnName ${matchColumnName}`,
+      )
+    } else if ('baseURL' in cardLink) {
+      const { baseURL, URLColumnName } = cardLink
+      const value = overrideValueWithRowID ? `syn${rowId}` : data[indexInData]
+      if (value) {
+        // value is defined!
+        return `/${baseURL}?${URLColumnName}=${encodeURIComponent(value)}`
+      }
+    }
+  }
+  return undefined
+}
+
+export function getLinkParams(
+  link: string,
+  cardLinkConfig: CardLink | undefined,
+  data: string[] | undefined,
+  schema?: Record<string, number>,
+  rowId?: number,
+) {
+  link = link.trim()
+  let href = link
+  const doiLink = convertDoiToLink(href)
+  let defaultTarget = TargetEnum.CURRENT_WINDOW
+  if (link.match(SYNAPSE_ENTITY_ID_REGEX)) {
+    // its a synId
+    href = `${PRODUCTION_ENDPOINT_CONFIG.PORTAL}Synapse:${link}`
+  } else if (doiLink) {
+    defaultTarget = TargetEnum.NEW_WINDOW
+    href = doiLink
+  } else if (!cardLinkConfig) {
+    defaultTarget = TargetEnum.NEW_WINDOW
+  } else if (cardLinkConfig) {
+    href = getCardLinkHref(cardLinkConfig, data, schema, rowId) ?? ''
+    if (href.includes('/DetailsPage')) {
+      defaultTarget = TargetEnum.NEW_WINDOW
+    }
+  }
+
+  const target = cardLinkConfig?.target ?? defaultTarget
+  return { href, target }
+}
+
+/**
+ * Maps a table row to a GenericCard.
+ */
+export function TableRowGenericCard(props: TableRowGenericCardProps) {
```

Comment:
Captures table-to-card mapping logic that was in the old GenericCard.

Once everything is mapped, props are passed to the new GenericCard